src/components/LegacyDids/LegacyDids.tsx

```diff
@@ -0,0 +1,57 @@
+import { Modal } from 'react-dialog-polyfill';
+import { browser } from 'webextension-polyfill-ts';
+import { generatePath, Link } from 'react-router-dom';
+
+import { useEffect } from 'react';
+
+import * as styles from './LegacyDids.module.css';
+
+import { paths } from '../../views/paths';
+
+import { useLegacyDidIdentities } from '../../utilities/legacyDids/legacyDids';
+import { useBooleanState } from '../../utilities/useBooleanState/useBooleanState';
+
+export function LegacyDids(): JSX.Element | null {
+  const t = browser.i18n.getMessage;
+
+  const open = useBooleanState();
+
+  const legacyDidIdentities = useLegacyDidIdentities();
+
+  useEffect(() => {
+    open.set(Object.values(legacyDidIdentities).length > 0);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [legacyDidIdentities]);
+
+  return (
+    <Modal open={open.current} className={styles.overlay}>
```

Comment:
I think this way we will not have a bunch of invisible DOM elements all the time, and the performance would be better.

```suggestion
  if (!open.current) {
    return null;
  }
  
  return (
    <Modal open className={styles.overlay}>
```